libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -183,6 +183,13 @@ void ArrayUtils::copyArrayToStorage(ArrayType const& _targetType, ArrayType cons
 			// copy
 			if (sourceBaseType->category() == Type::Category::Array)
 			{
+				solUnimplementedAssert(
+					_sourceType.location() != DataLocation::CallData ||
+					_sourceType.isDynamicallySized() ||
```

Comment:
Are you sure this should not be negated?

Reply:
The type that makes problems is `uint[][2]`, so the outer array is a non-dynamically sized one

Reply:
But `uint[][]` would not work either, would it?

Reply:
`uint[][]` is dynamically sized, so that does work